### DISCUSSION TITLE
Policy based on domain name

### DISCUSSION THREAD

**Discussion by itechnica** [supabase/supabase#2233] [Questions] [answered]
Is it possible for policies to restrict access based on the referring domain?

e.g. if an API request is made from thedomain.com or example.otherdomain.com, it would only return records containing a specific column that contains the referring fully-qualified domain name?

I suppose the question is, is it possible for a policy to access the refer in the request header of an API request?

The use case is for multi-tenancy service, where query results are only returned for the appropriate organisation (based on the fully qualified domain name).

Hopefully, I've got my terminology right!

Thank you.

**Comment by steve-chavez** [comment, marked as answer]
Hey @itechnica,

> I suppose the question is, is it possible for a policy to access the refer in the request header of an API request?

Yes, it can be accessed by doing:

```sql
SELECT current_setting('request.header.referrer', true)
```

More details at https://postgrest.org/en/stable/api.html#accessing-request-headers-cookies-and-jwt-claims

**Reply by itechnica** [reply]
Thank you very much for taking the time to reply with the reference; I was able to implement a policy as required.

For anyone who needs an example, your policy might look like this:
```
(
  current_setting('request.header.origin':: text, true) = 'https://example.domain.com':: text
)
```
